### PR TITLE
harden(scoreboard): fail-open on corrupt scoreboard/cooldown/wave + enforce result vocabulary

### PR DESCRIPTION
**Pull request by jobordu**
## What

`/nf:harden` on `bin/update-scoreboard.cjs` — the quorum's **persistent memory** (per-slot success rates + cooldown windows). It feeds `nf-prompt`'s slot ordering and the availability filter, so a bug here silently skews *future* roster decisions. Two adversarial agents + a regression round; converged over 2 rounds. **7 reachable gaps**, all the error-path shape-guard / validation-enforcement class.

## 🔴 Gaps fixed

- **`loadData` didn't degrade wrong-shape JSON** — it restored only 3 of 5 top-level fields and never sanitised rounds, so a `{}` / missing-`models` / non-array-`rounds` / null-round scoreboard crashed the record path (`data.models[model]` / `findIndex(r => r.task)`) → exit 1, **vote lost, file left corrupt**. Now normalizes all fields and filters non-object rounds.
- **`models: []` slipped the normalization** (round-2 find) — `typeof [] === 'object'`, so an array-shaped field passed as an object; the record path then wrote a non-index prop that `JSON.stringify` drops → **stats silently lost on write** (CLI even printed the right score, masking it). Now excludes arrays for every object field.
- **merge-wave declared `VALID_RESULTS` but never enforced it** — the **#268 pattern recurring in the same file**: a bogus result code was persisted verbatim and **inflated the per-slot invocation denominator** while scoring nothing. Now enforced against the canonical vocabulary (dead local Set removed).
- **merge-wave: one bad file aborted the whole atomic transaction** — a wrong-shape (`null`) wave file threw → exit 1 *before* the single write → **every valid sibling vote in the wave lost**. Now shape-guarded per file (skip, don't abort).
- **get-availability: stuck-in-cooldown forever** — a corrupt `available_at_iso` made `is_available = (NaN <= now) = false` → the slot **silently excluded from the quorum forever** with NaN remaining. Now fails open (unparseable window = available now).
- **set-availability: crash** — an absurd cooldown hint → Invalid Date → `toISOString()` `RangeError` → exit 1, cooldown never recorded. Now guarded.

## Verification (2 rounds, every fix red-proven)

- **Round 1** (2 agents): 6 gaps fixed (recording/recompute + availability/merge-wave).
- **Round 2**: regression matrix confirmed the fixes don't over-correct — populated scoreboards round-trip intact, every canonical result code (incl. `TP+`/`TN+`/Mode-A `""`) is still accepted, a genuine future cooldown is still benched — and caught the `models: []` miss.

**+3 adversarial test files wired into `test:ci`. Per-file all green; combined 72/72** (one transient parallel-`test:ci` flake confirmed clean on re-run). lint OK; installed copy in sync.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved scoreboard loading to handle partially malformed data more safely.
  * Availability checks now avoid invalid time calculations and treat bad timestamps as immediately available.
  * Recording and merge flows now skip invalid vote entries and reject unsupported result codes without crashing.
  * Preserved existing data more reliably when re-recording or updating rounds.

* **Tests**
  * Added new adversarial regression tests covering malformed scoreboards, availability edge cases, and recording robustness.
  * Expanded CI test coverage to run the new adversarial suites.

### CODE REVIEW

File: bin/update-scoreboard-availability-adversarial.test.cjs [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+    ]);
+
+    assert.notStrictEqual(
+      exitCode, 1,
+      `set-availability must not FATAL-crash on an absurd duration; stderr was: ${stderr}`
+    );
+    assert.ok(
+      !/Invalid time value/.test(stderr),
+      `must not throw "Invalid time value"; stderr was: ${stderr}`
+    );
```
[the remark this answers is not in the document]
Comment (jobordu) [reply bin/update-scoreboard-availability-adversarial.test.cjs:206]: Fixed in the latest commit — AVAIL-ADV-3 now asserts strictEqual(exitCode, 0) (genuine graceful success: the fix returns exit 0 and skips the bad window), so a timeout or signal kill (exitCode null) no longer passes.
